transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -218,6 +220,12 @@ public B validate() {
         return (B) this;
     }
 
+    private void ensureChannelFactory() {
+        if (channelFactory == null) {
+            channelFactory = ChannelFactoriesRegistry.getFactoryForEventLoopGroup(group.getClass(), isServerSide());
```

Comment:
@rkapsi IMHO using "instanceof" in a parent class on 'this' is not a good object oriented design choice.
But I will change it if you confirm it is better for the rules of the project